Emit dots on travis to avoid timeout

TheRealFalcon opened the pull request:
## Proposed Commit Message
<!-- Include a proposed commit message because all PRs are squash merged -->

```
Emit dots on travis to avoid timeout

The current method of running a background sleep until travis is
finished is causing integration test runs to pass even when they should
be failing.

Instead, update the code to emit dots itself.
```

## Additional Context
<!-- If relevant -->

## Test Steps
Look for dots during launch of tests on travis (if operation takes more than 10 seconds). Everything else should be the same.

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes
that apply. -->
 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

smoser commented:
Could you use [travis_wait](https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received) ?

TheRealFalcon commented:
Yeah, we discussed that option too. It waits until the end to give you any output, and I think people prefer to get output as it happens.

TheRealFalcon commented:
Removed the WIP title. I have verified that failures actually fail the build now. I also verified that we get dots on . For an LXD VM build, we get something like:
```
2021-04-14 18:58:12 INFO      integration_testing:clouds.py:168 Launching instance with launch_kwargs:
image_id=ubuntu:40513f27bbe99e53c254ee7f149ac14835e6e5bcf588096f1b3c3b2e41e79146
user_data=None
2021-04-14 18:58:12 INFO      integration_testing:utils.py:41 .
2021-04-14 18:58:42 INFO      integration_testing:utils.py:41 .
2021-04-14 18:59:12 INFO      integration_testing:utils.py:41 .
2021-04-14 18:59:29 INFO      pycloudlib.instance:instance.py:165 executing: sh -c whoami
2021-04-14 18:59:42 INFO      integration_testing:utils.py:41 .
2021-04-14 19:00:12 INFO      integration_testing:utils.py:41 .
2021-04-14 19:00:14 INFO      pycloudlib.instance:instance.py:343 [Errno None] Unable to connect to port 22 on 10.113.228.112
Retrying SSH connection to ubuntu@10.113.228.112:22 (555s left)
2021-04-14 19:00:15 INFO      pycloudlib.instance:instance.py:343 [Errno None] Unable to connect to port 22 on 10.113.228.112
Retrying SSH connection to ubuntu@10.113.228.112:22 (554s left)
2021-04-14 19:00:17 INFO      pycloudlib.instance:instance.py:343 [Errno None] Unable to connect to port 22 on 10.113.228.112
Retrying SSH connection to ubuntu@10.113.228.112:22 (552s left)
2021-04-14 19:00:21 INFO      pycloudlib.instance:instance.py:343 [Errno None] Unable to connect to port 22 on 10.113.228.112
Retrying SSH connection to ubuntu@10.113.228.112:22 (548s left)
2021-04-14 19:00:22 INFO      paramiko.transport:transport.py:1819 Connected (version 2.0, client OpenSSH_8.2p1)
```
pytest swallows print statements, so I had to do it via logging. Newlines aren't ideal, but neither is changing our code for travis :stuck_out_tongue:

@OddBloke I renamed `log_utils.py` to `utils.py`. Feel free to shame me for it.

OddBloke commented on tests/integration_tests/utils.py:
```diff
@@ -0,0 +1,49 @@
+import logging
```
Shall we go with `util.py` for consistency with ~~our other~~`cloudinit.util`?

"Consistency with our other util files":

```
$ find cloudinit -name \*util.py
cloudinit/ssh_util.py
cloudinit/sources/helpers/vmware/imc/guestcust_util.py
cloudinit/tests/test_util.py
cloudinit/util.py
cloudinit/distros/ug_util.py
cloudinit/distros/rhel_util.py
cloudinit/distros/net_util.py
$ find cloudinit -name \*utils.py
cloudinit/temp_utils.py
cloudinit/type_utils.py
cloudinit/cs_utils.py
cloudinit/ec2_utils.py
cloudinit/tests/test_temp_utils.py
cloudinit/distros/bsd_utils.py
```

OddBloke reviewed (approved):
LGTM, thanks! (I'll leave it up to you whether you rename the file before landing.)
